src/lib/routes/admin-api/user-admin.ts

```diff
@@ -716,4 +725,28 @@ export default class UserAdminController extends Controller {
             projectRoles,
         });
     }
+
+    async throwIfScimUser({
+        id,
+        scimId,
+    }: Pick<IUser, 'id' | 'scimId'>): Promise<void> {
+        if (this.isEnterprise && this.flagResolver.isEnabled('scimApi')) {
+            const { enabled } = await this.settingService.getWithDefault(
+                'scim',
+                { enabled: false },
+            );
+
+            if (enabled) {
+                const isScim =
+                    Boolean(scimId) ||
+                    Boolean((await this.userService.getUser(id)).scimId);
+
+                if (isScim) {
+                    throw new ForbiddenError(
+                        'Cannot perform this operation on SCIM users',
```

Comment:
Tempted to make this a little... dumber. We're knee deep in SCIM so we know what this is. But an arbitrary user may not have all the context (they probably should since they were probably the one that set it up but that's life).

What about "This user is managed by your SCIM provider and cannot be changed manually"?

Not going to put a suggestion because that's gonna set fire to your tests O.O

Reply:
Legend!